src/WrappedMToken.sol

```diff
@@ -689,6 +688,48 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
         emit StoppedEarning(account_);
     }
 
+    /**
+     * @dev   Transfer `amount_` M to `recipient_`, tracking this contract's M balance rounding errors.
+     * @param recipient_ The account to transfer M to.
+     * @param amount_    The amount of M to transfer.
+     */
+    function _transferM(address recipient_, uint240 amount_) internal {
+        uint240 startingBalance_ = _mBalanceOf(address(this));
+
+        // NOTE: The behavior of `IMTokenLike.transfer` is known, so its return can be ignored.
+        IMTokenLike(mToken).transfer(recipient_, amount_);
+
+        // NOTE: When this WrappedMToken contract is earning, any amount of M sent from it is converted to a principal
+        //       amount at the MToken contract, which when represented as a present amount, may be a rounding error
+        //       amount more than `amount_`. In order to capture the real decrease in M, the difference between the
+        //       ending and starting M balance is returned.
+        uint240 decrease_ = startingBalance_ - _mBalanceOf(address(this));
+
+        // If the M lost is more than the wM burned, then the difference is added to `roundingError`.
+        roundingError += int144(int256(uint256(decrease_)) - int256(uint256(amount_)));
+    }
+
+    /**
+     * @dev   Transfer `amount_` M from `sender_`, tracking this contract's M balance rounding errors.
+     * @param sender_ The account to transfer M from.
+     * @param amount_ The amount of M to transfer.
+     */
+    function _transferFromM(address sender_, uint240 amount_) internal {
+        uint240 startingBalance_ = _mBalanceOf(address(this));
+
+        // NOTE: The behavior of `IMTokenLike.transferFrom` is known, so its return can be ignored.
+        IMTokenLike(mToken).transferFrom(sender_, address(this), _getSafeTransferableM(sender_, amount_));
```

Comment:
Perhaps the answer to my question previous question is related to the use of `getSafeTransferableM` here. Since the logic above also seems to apply to the error below otherwise.

Reply:
In some tests I noticed that even when pulling a "safe transferable amount" from an earner, it is still possible that the principal amount results in the recipient receiving some rounding error more.